src/Interpreters/FillingRow.h

```diff
@@ -17,7 +18,25 @@ bool equals(const Field & lhs, const Field & rhs);
 class FillingRow
 {
 public:
-    FillingRow(const SortDescription & sort_description);
+    struct
+    {
+        FillingRow & filling_row;
+
+        Field & operator[](size_t index) { return filling_row.row[index]; }
+        const Field & operator[](size_t index) const { return filling_row.row[index]; }
+        size_t size() const { return filling_row.sort_description.size(); }
+    } sort;
+
+    struct
+    {
+        FillingRow & filling_row;
+
+        Field & operator[](size_t index) { return filling_row.row[filling_row.sort_description.size() + index]; }
+        const Field & operator[](size_t index) const { return filling_row.row[filling_row.sort_description.size() + index]; }
+        size_t size() const { return filling_row.interpolate_description.size(); }
+    } interpolate;
```

Comment:
Can we just split `Row row` into two parts : `sort_row` and `interpolate_row`, and avoid this indexation? It's a little bit hard to read.

We can probably remove `operator[]` and `size()` methods, and add `getSortRow/getInterpolateRow` instead.